File: packages/contracts-ts/src/instrument.ts
import { z } from "zod";
import { AssetClass } from "./common";

export const Exchange = z.object({
  code: z.string(),
  name: z.string(),
  mic: z.string().nullable().optional(),
  timezone: z.string().default("Europe/Istanbul"),
});
export type Exchange = z.infer<typeof Exchange>;

export const Instrument = z.object({
  tenant_id: z.string().default("default"),
  symbol: z.string(),
  exchange_code: z.string().default("XIST"),
  name: z.string(),
  asset_class: AssetClass.default("equity"),
  currency: z.string().default("TRY"),
  yfinance_ticker: z.string().nullable().optional(),
  isin: z.string().nullable().optional(),
  is_active: z.boolean().default(true),
});
export type Instrument = z.infer<typeof Instrument>;

File: packages/contracts-py/heimdall_contracts/instrument.py
"""Enstrüman ve borsa kontratları."""

from __future__ import annotations

from pydantic import Field

from .common import AssetClass, HeimdallModel


class Exchange(HeimdallModel):
    code: str = Field(description="Dahili borsa kodu, ör. 'XIST'.")
    name: str = Field(description="Görünen ad, ör. 'Borsa İstanbul'.")
    mic: str | None = Field(default=None, description="ISO 10383 MIC, ör. 'XIST'.")
    timezone: str = "Europe/Istanbul"


class Instrument(HeimdallModel):
    tenant_id: str = "default"
    symbol: str = Field(description="Borsa sembolü, ör. 'THYAO'.")
    exchange_code: str = "XIST"
    name: str = Field(description="Şirket/enstrüman adı.")
    asset_class: AssetClass = AssetClass.EQUITY
    currency: str = "TRY"
    yfinance_ticker: str | None = Field(
        default=None, description="yfinance sembolü, ör. 'THYAO.IS'."
    )
    isin: str | None = None
    is_active: bool = True
